test/integration/good/code-gen/cpp.expected

```diff
@@ -1215,311 +1217,329 @@ static constexpr std::array<const char*, 427> locations_array__ =
  " (in 'complex_scalar.stan', line 34, column 4 to column 13)",
  " (in 'complex_scalar.stan', line 33, column 45 to line 35, column 3)"};
 
-
-std::complex<double>
-foo(std::ostream* pstream__) {
-  using local_scalar_t__ = double;
-  int current_statement__ = 0; 
-  static constexpr bool propto__ = true;
-  (void) propto__;
-  local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
-  (void) DUMMY_VAR__;  // suppress unused var warning
-  try {
-    current_statement__ = 405;
-    return stan::math::to_complex();
-  } catch (const std::exception& e) {
-    stan::lang::rethrow_located(e, locations_array__[current_statement__]);
-  }
-  
-}
-
+struct foo8_functor__ {
+template <typename T0__> stan::promote_args_t<T0__>
+  operator()(const std::vector<std::vector<std::complex<T0__>>>& z,
+             std::ostream* pstream__) const;
+};
+struct foo1_functor__ {
+template <typename T0__> stan::promote_args_t<T0__>
+  operator()(const std::complex<T0__>& z, std::ostream* pstream__) const;
+};
+struct foo4_functor__ {
+std::vector<std::complex<double>>
+  operator()(std::ostream* pstream__) const;
+};
+struct foo6_functor__ {
+template <typename
+  T0__>
+  std::vector<std::complex<stan::promote_args_t<T0__>>>
```

Comment:
Can this be pretty printed to remove the newline?
```suggestion
template <typename T0__>
  std::vector<std::complex<stan::promote_args_t<T0__>>>
```

Reply:
I spent a lot of time wrestling with the pretty printer. Let me take another pass at it.


A long-term project (likely after tuples) I'm thinking about is defining a proper LIR which maps to C++ so we don't have to do all this string generation, just call `code_gen: MIR -> LIR` followed py `pp_lir: LIR -> str`

Reply:
I've played around with the C++ output, how does it look in the latest commit?

Reply:
Good!